Lib/pdb.py

```diff
@@ -206,6 +207,10 @@ def namespace(self):
 line_prefix = '\n-> '   # Probably a better default
 
 class Pdb(bdb.Bdb, cmd.Cmd):
+    # the max number of chained exceptions + exception groups we accept to navigate.
+    _max_chained_exception_depth = 999
+    _chained_exceptions = tuple()
+    _chained_exception_index = 0
```

Comment:
Why do we put these in the class? Is there any case `_chained_exceptions` and `_chained_exception_index` would be used without proper initialization in the instances? `previous_sigint_handler` is used as `Pdb._previous_sigint_handler` and it's probably set for all `Pdb` instances, but the exceptions are per instance right? I think we should init it as instance members in `__init__`.

As for `_max_chained_exception_depth`, if that's an arbitrary constant without a documented way to change, we can consider that as a constant and use `MAX_CHAINED_EXCEPTION_DEPTH`?

Reply:
I rarely see multiple Pdb instances used so I don't have any particular preferences. 

Though I don't quite like global variables, and I have the impression that attaching constant to the only class that use them make things a tiny bit more self contained and discoverable.

I still did both of the changes you suggested though.

Reply:
I obviously misrepresented myself. I meant use `MAX_CHAINED_EXCEPTION_DEPTH` as a class member instead of `_max_chained_exception_depth`. Just change the variable name to indicate that this is a constant instead of a private variable. I still want to put it in `class Pdb` because that's the only scope where it would be used.

Reply:
Ah, sure, I moved it back as a class variable.